.evergreen.yaml

```diff
@@ -68,12 +74,43 @@ functions:
       params:
         ignore_missing_file: true
         file: src/atlas-service-broker/version_expansion.yaml
+
   "build_binaries":
     - command: shell.exec
       params:
         working_dir: src/atlas-service-broker
         script: |
           ./dev/scripts/build-production-binary.sh artifacts/atlas-service-broker-linux-arm64-${version}
+
+  # Upload integration test binary to S3. MMS will use this binary as part of the Atlas test suite.
+  "upload_test_binary":
+    - command: shell.exec
+      params:
+        working_dir: src/atlas-service-broker
+        script: |
+          # Output build info to metadata file
+          echo '{ "master": { "int_test": "artifacts/${revision}/int-test", "git_hash": "${revision}" } }' > builds.json
+
+          go test -v -timeout 1h -c -o artifacts/${revision}/int-test ./test/integration
+    - command: s3.put
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        local_file: src/atlas-service-broker/builds.json
+        remote_file: atlas-service-broker/builds.json
+        bucket: mciuploads
+        content_type: application/json
+        permissions: public-read
+    - command: s3.put
+      params:
```

Comment:
better to upload the binary successfully before updating the metadata

Reply:
Good point, fixed!